### PR TITLE
UIREC-20 Receiving: Access the Piece Details Wizard

### PR DESCRIPTION
**Pull request by AndreiShumski2**
## Purpose
To make PO line pieces details modal as a wizard with clickable Next/Previous buttons
https://issues.folio.org/browse/UIREC-20

### To Do
Tests for `ItemDetails` component will be written after updating `ReceivingList`

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/components/Receiving/ItemDetailsFooter.js [hunk 1]
```diff
@@ -7,30 +7,58 @@ import {
...
-    </Button>
-    <Button
-      buttonStyle="primary"
-      disabled
-    >
-      <FormattedMessage id="ui-orders.receiving.nextBtn" />
-    </Button>
-  </Row>
-);
+const ItemDetailsFooter = props => {
```
Comment (aliaksei-chumakou) [inline src/components/Receiving/ItemDetailsFooter.js]:
```
const ItemDetailsFooter = ({
  prop1,
  prop2,
}) => (
  <Row end="xs">
  </Row>
);
```

File: src/components/Receiving/ItemDetails.js [hunk 2]
```diff
@@ -31,61 +31,85 @@ class ItemDetails extends Component {
...
     });
   }
 
-  render() {
-    const { close, linesItemList, locationsOptions } = this.props;
-    const poLineNumber = get(linesItemList, [0, 'poLineNumber'], '');
-    const title = get(linesItemList, [0, 'title'], '');
-    const resultsFormatter = {
+  onClickControlButton = (type) => {
+    switch (type) {
```
Comment (aliaksei-chumakou) [inline src/components/Receiving/ItemDetails.js]: IMHO, for this kind of switch there is no too much sense, it looks like over-engineering. Just create 2 function - `clickNext` and `clickPrevious` and pass them to footer.

File: src/components/Receiving/ItemDetails.js [hunk 3]
```diff
@@ -31,61 +31,85 @@ class ItemDetails extends Component {
...
           checked={this.isItemChecked(item)}
         />
       ),
-      'barcode': () => (
+      'barcode': (item) => (
         <div className={css.editableFieldWrapper}>
-          <TextField type="number" />
+          <TextField
+            type="number"
+            value={get(item, 'barcode', '')}
```
Comment (aliaksei-chumakou) [inline src/components/Receiving/ItemDetails.js]: you set value here, what's about `onChange` attribute? User should be able to enter barcode.

File: src/components/Receiving/ItemDetails.js [hunk 4]
```diff
@@ -31,61 +31,85 @@ class ItemDetails extends Component {
...
         </div>
       ),
       'format': (item) => <KeyValue value={get(item, 'orderFormat', '')} />,
-      'location': () => (
+      'location': (item) => (
         <div className={css.editableFieldWrapper}>
           <Select
             dataOptions={locationsOptions}
-            defaultValue=""
+            defaultValue={get(item, 'locationId', '')}
```
Comment (aliaksei-chumakou) [inline src/components/Receiving/ItemDetails.js]: I think defaultValue shouldn't be here, try to pass locationId to `value`

File: src/components/Receiving/ItemDetails.js [hunk 5]
```diff
@@ -31,61 +31,85 @@ class ItemDetails extends Component {
     super(props);
 
     this.state = {
-      allChecked: false,
+      allChecked: {},
       checkedItems: [],
+      currentLine: 0,
+      lineItems: this.props.linesItemList,
```
Comment (aliaksei-chumakou) [inline src/components/Receiving/ItemDetails.js]: I don't see any reason to do that. Do you?

Comment (AndreiShumski2) [reply src/components/Receiving/ItemDetails.js]: To update barcode, locationId.

**Pull request merged**